### PR TITLE
Jobseeker notified job revoked 481

### PR DESCRIPTION
**Pull request by vijayakarumudi**
Closes AgileVentures/MetPlus_tracker#481

### CODE REVIEW

**Comment by vijayakarumudi** [comment]
PR for review

File: app/models/event.rb [hunk 1]
```diff
@@ -426,6 +426,26 @@ def self.evt_job_revoked(evt_obj)
...
+                     EVT_TYPE[:JOB_REVOKED],
+                     job_id:       evt_obj.job.id,
+                     job_title:    evt_obj.job.title,
+                     company_name: evt_obj.job.company.name,
+                     notify_list:  js_ids)
+
+      NotifyEmailJob.set(wait: delay_seconds.seconds)
+                    .perform_later(js_emails,
+                                   EVT_TYPE[:JOB_REVOKED],
+                                   evt_obj.job)
```
Comment (patmbolger) [inline app/models/event.rb]: This is using the mailer for intra-agency emails (that is, email that is sent only to agency people).

It needs to use the mailer for job seekers.  

Changes required:

1. Change `NotifyEmailJob` in above statement to `JobSeekerEmailJob`

2. Add code to handle that event to `job_seeker_email_job.rb`

3. Add method to handle that email to `job_seeker_email.rb`

4. Add template for the email to `views/job_seeker_mailer`

The email body should include something like:
```
A job that you had applied to has been removed from PETS by the company which posted that job.

The job is: <job_title>, and was posted by <company name>  << job_title is a link to the job posting

You had applied to this job on <date>  << e.g. `You had applied to this job on Tuesday, November 1, 2016.

If you have any questions about this email or the use of PETS
  then please contact the PETS administrator at:
  = mail_to("#{ENV['NOTIFICATION_EMAIL']}", 'PETS Admin',
         subject: 'Job revoked email notification')
```

Comment (vijayakarumudi) [inline app/models/event.rb]: I am working on these changes.

**Comment by patmbolger** [comment]
Please remember to add a preview for this email to `spec/mailers/previews`

**Comment by patmbolger** [comment]
Please remember to change the title of this PR to comply with our naming standard.

File: app/models/event.rb (hunk 1 above)
Comment (vijayakarumudi) [inline app/models/event.rb]: I completed all the change requests. I will  update the PR tomorrow.

**Comment by joaopapereira** [comment]
@vijayakarumudi Can you please rebase the PR?

**Comment by vijayakarumudi** [comment]
Updated PR for review.

File: spec/mailers/previews/job_seeker_mailer_preview.rb [hunk 2]
```diff
@@ -18,8 +17,19 @@ def job_applied_by_job_developer
     job_developer = User.find_by_email('chet@metplus.org').actable
     company       = Company.find_by_email('contact@widgets.com')
     job           = company.jobs.first
-  
+
     JobSeekerMailer.job_applied_by_job_developer(job_seeker, job_developer, job)
   end
 
+  def job_revoked
+    job_seeker = User.find_by_email('tom@gmail.com').actable
```
Comment (patmbolger) [inline spec/mailers/previews/job_seeker_mailer_preview.rb]: This line fails - the seeded record that we want has an email of `tomseekerpets@gmail.com`.

File: spec/mailers/previews/job_seeker_mailer_preview.rb [hunk 3]
```diff
@@ -18,8 +17,19 @@ def job_applied_by_job_developer
...
+    job = Job.create(title: 'Sr Software Engineer',
+                     company: Company.first,
+                     company_job_id: 'ABC',
+                     shift: Job::SHIFT_OPTIONS[0],
+                     description: 'description of test job')
+    FactoryGirl.create(:resume, job_seeker: job_seeker)
```
Comment (patmbolger) [inline spec/mailers/previews/job_seeker_mailer_preview.rb]: This should not be needed as the seeded JS has an associated résumé.

File: app/models/event.rb [hunk 4]
```diff
@@ -426,6 +426,26 @@ def self.evt_job_revoked(evt_obj)
...
+                     EVT_TYPE[:JOB_REVOKED],
+                     job_id:       evt_obj.job.id,
+                     job_title:    evt_obj.job.title,
+                     company_name: evt_obj.job.company.name,
+                     notify_list:  js_ids)
+
+      JobSeekerEmailJob.set(wait: delay_seconds.seconds)
+                       .perform_later(js_emails,
+                                      EVT_TYPE[:JOB_REVOKED],
+                                      evt_obj.job)
```
Comment (patmbolger) [inline app/models/event.rb]: This is not actually sending any emails as the calling signature is not correct.

The JobSeekerMailer method is not defined to accept an array of email addresses.

The call to the job should look more like this:
```
JobSeekerEmailJob.set(wait: delay_seconds.seconds)
                 .perform_later(EVT_TYPE[:JOB_REVOKED],
                                ja.job_seeker,
                                evt_obj.job)
```
And that call should be wrapped in a loop that cycles through all job seekers.

File: spec/mailers/previews/job_seeker_mailer_preview.rb (hunk 2 above)
Comment (vijayakarumudi) [inline spec/mailers/previews/job_seeker_mailer_preview.rb]: "tomseekerpets@gmail.com"  generated nil class. So I kept existing email.

**Comment by vijayakarumudi** [comment]
close

**Comment by vijayakarumudi** [comment]
closed PR

**Comment by vijayakarumudi** [comment]
open PR

**Comment by vijayakarumudi** [comment]
Reopen PR

**Pull request closed without merging**